elfi/methods/diagnostics.py

```diff
@@ -0,0 +1,294 @@
+"""Methods for ABC diagnostics."""
+
+import logging
+from itertools import combinations
+
+import numpy as np
+from scipy.spatial import cKDTree
+from scipy.special import digamma, gamma
+
+import elfi
+
+logger = logging.getLogger(__name__)
+
+
+class TwoStageSelection:
+    """Perform the summary-statistics selection proposed by Nunes and Balding (2010).
+
+    The user can provide a list of summary statistics as list_ss, and let ELFI to combine them,
+    or provide some already combined summary statistics as prepared_ss.
+
+    The rationale of the Two Stage procedure procedure is the following:
+    - First, the module computes or accepts the combinations of the candidate summary statistics;
+    - In Stage 1, each summary-statistics combination is evaluated using the
+    Minimum Entropy algorithm;
+    - In Stage 2, the minimum-entropy combination is selected,
+    and the `closest' datasets are identified;
+    - Further in Stage 2, for each summary-statistics combination,
+    the mean root sum of squared errors (MRSSE) is calculated over all `closest datasets',
+    and the minimum-MRSSE combination is chosen as the one with the optimal performance.
+
+    Attributes
+    ----------
+    simulator : elfi.Node
+        Node (often elfi.Simulator) for which the summary statistics will be applied.
+        The node is the final node of a coherent ElfiModel (i.e. it has no child nodes).
+    fn_distance : str or callable function
+        Function for calculating distance (can be a string if the function is implemented in ELFI).
+    list_ss : List of callable functions, optional
+        List of uncombined candidate summary statistics.
+    prepared_ss : List of lists of callable functions, optional
+        List of combined candidate summary statistics.
```

Comment:
'Prepared combinations of candidate summary statistics. No other combinations will be evaluated.'?

Reply:
Renamed as suggested.